doc/rds-day-2.md

```diff
@@ -0,0 +1,153 @@
+
+In this approach, it is important for the check mechanisms to work reliably. It's easy to lose developers' trust when checks say there will be restarts but no restart happened. Or worse, checks did not detect potential restarts and caused an outage.
+
+### Check at runtime
+
+#### Approach 1 
+Another approach is to deny such operation at runtime using a policy engine and/or custom validating web hook unless certain conditions are met. This means problems with RDS configuration is communicated to the developers through their GitOps tooling by providing reasons for denial.
+
+```mermaid
+flowchart LR
+    subgraph Kubernetes
+        ConfigMap(ConfigMap w/ ticket numbers)
+        ValidatingController(Policy Engine / Validating controller)
+    end 
+
+    subgraph Git 
+        PR(PR Merged)
+    end
+
+    subgraph Ticketing
+       Approved(Approved Changes)
+    end
+    
+    GitOps(GitOps tooling)
+
+    PR(PR Merged) --> GitOps --> ValidatingController
+    ValidatingController --reference--> ConfigMap
+    ValidatingController --deny and provide reason--> GitOps
+    Approved --push--> ConfigMap
+```
+In the example above, no check is performed during PR. During admission into the Kubernetes cluster, a validating controller will lookup config map which contain ticket number and validate the request is valid. If no ticket number associated with this change is approved, it's rejected with provided reason. 
+
+#### Approach 2
+
+```mermaid
+flowchart LR
+    subgraph Kubernetes
+        ConfigMap(ConfigMap w/ ticket numbers)
+        ValidatingController(Policy Engine / Validating controller)
+    end 
+
+    subgraph Git
+        subgraph PR
+            Claim
+            Manifests(Other Manifests)
+        end
+    end
+
+    subgraph Ticketing
+       Approved(Approved Changes)
+    end
+    
+    User
+    GitOps(GitOps tooling)
+
+    User --Create Ticket--> Ticketing
+    User --Annoate with ticket number--> Claim
+    PR(PR Merged) --> GitOps --> ValidatingController
+    ValidatingController --reference--> ConfigMap
+    ValidatingController --deny if not approved \n and provide reason--> GitOps
+    Approved --create when the ticket \n is approved--> ConfigMap
+    
+```
+In this example, developer creates a ticket in the ticketing system and annotates the infrastructure claim with the ticket number. The admission controller checks if the change affects fields that require approval. If approval is required, the change is denied until the ticket is approved and the reason is given back to the GitOps tooling. 
+
+Once the ticket is approved, a config map is created with the ticket number as its name or as one of annotations. Next time the GitOps tooling attempts to apply manifests, the admission controller sees the ConfigMap is now created and allows it to be deployed. Once it is deployed, the ConfigMap can be marked for deletion. In both approaches, there is no need for read access to the ticketing system. 
```

Comment:
this seems to require so much machinery. is it worth offering a solution here?

Reply:
this mermaid thingy is pretty cool btw. first time I am seeing it in a doc.

Reply:
I think there's a value here to show different approaches. Different organizations have different requirements so something like this may make sense for some.

Reply:
My assumption is that an organizations would use above process to "fit" into already existing audit compliance process. An alternative, and my recommendation, would be to augment the audit compliance process instead to include the track record in the git repo.